dvc/render/converter/vega.py

```diff
@@ -233,7 +232,8 @@ def flat_datapoints(self, revision):  # noqa: C901
         for i, (y_file, y_field) in enumerate(ys):
             if num_xs > 1:
                 x_file, x_field = xs[i]
-            datapoints = deepcopy(file2datapoints.get(y_file, []))
+
+            datapoints = [d.copy() for d in file2datapoints.get(y_file, [])]
```

Comment:
file2datapoints seems to be a list of flat dictionaries. So, instead of doing a deepcopy, we can iterate and do a shallow copy instead. 

Let me know if these assumptions are incorrect.

```python
[
  {'step': '0', 'train/acc': '0.2707333333333333'},
  {'step': '1', 'train/acc': '0.40696666666666664'},
]
```